packages/gui/src/components/walletConnect/WalletConnectAddConnectionDialog.tsx

```diff
@@ -118,6 +120,13 @@ export default function WalletConnectAddConnectionDialog(props: WalletConnectAdd
     }
 
     const topic = await pair(uri, selectedFingerprints, mainnet);
+    if (bypassCheckbox) {
+      let skipReadOnlyObject = {};
+      selectedFingerprints.forEach((f: number) => {
+        skipReadOnlyObject = { ...skipReadOnlyObject, [f]: true };
+      });
+      setBypassReadonlyCommands({ ...bypassReadonlyCommands, [topic.toString()]: skipReadOnlyObject });
```

Comment:
It seems odd to create a nested data structure to hold a boolean. The data structure could be simplified to:
```
{
  <topic_1>: [fingerprint_1, fingerprint_2],
  <topic_2>: [fingerprint_3]
}
```

If the fingerprint is in the list, bypass confirmation is enabled. If it's not, then confirmation is required.